plugins/outputs/zabbix/README.md

```diff
@@ -0,0 +1,396 @@
+Telegraf
+- it will also send the traps to fill the new created items
+
+### Reducing the number of LLDs
+
+This plugin remembers which LLDs has been sent to Zabbix and avoid generating
+the same metrics again, to avoid the cost of LLD processing in Zabbix.
+
+It will only send LLD data each `lld_period` intervals of this output plugin.
+
+For example, if the `flush_interval` for this plugins is set to `60s` and
+`lld_period` set to 10, each 10 minutes Telegraf will send a metric to the
+discovery rules, but only if new data has been seen.
+
+But, could happen that package is lost or some host get new discovery rules, so
+each `lld_reset_period` the plugin will forget about the known data and start
+collecting again.
+
+For example, given `flush_interval=60s`, `lld_period=10` and
+`lld_reset_period=6`:
+
+- the first 10 minutes this plugin will collect information about measurements
+and tags
+- in the 10th flush it will send the LLD data
+- in the next 10 min, it will keep collecting data
+- in the 20th flush, it will only send LLD data if new data has been seen
+- it will repeat this pattern till the 50th flush
+- from the 50th flush to the 60th flush it will collect new info, forgetting
+the already captured
+- the 60th flush will send LLD data seen between the 50th and the 60th flushes.
+
+## Configuration
+
+```toml @sample.conf
+# Send metrics to Zabbix
+[[outputs.zabbix]]
+  ## Address of zabbix host
+  host = "zabbix.example.com"
+
+  ## Port of the Zabbix server
+  port = 10051
+
+  ## Send metrics as type "Zabbix agent (active)"
+  agent_active = false
+
+  ## Add prefix to all keys sent to Zabbix (set by default)
+  prefix = "telegraf."
+
+  ## Skip measurement prefix to all keys sent to Zabbix (false by default)
+  skip_measurement_prefix = false
+
+  ## This field will be sent as HostMetadata to Zabbix Server to autoregister the host.
+  ## If this field is not defined, then no autoregistration will be done.
+  autoregister = "Example aaa222111cccdddaaaa"
+
+  ## This is the period with which self-registrations are sent.
+  ## Only applies if autoregister is defined in config file.
+  ## This period is for each hostname seen by Telegraf. After this period, the autoregistration for this
+  ## hostname will be sent again (one telegraf could handle several hostnames).
+  autoregister_send_period = "30m"
+
+  ## Number of writes (output flushes) after LLD data is sent.
+  ## If the output flush interval is set to 60s and lld_period is set to 10, then the LLD data will be sent every 10m.
+  ## Only new LLD data will be sent.
+  lld_period = 10
+
+  ## Numer of LLD sends after all LLDs are sent again.
+  ## This will flush known LLD data and start capturing it again.
+  ## Example, if flush interval is set to 60s, lld_period is set to 10 and lld_reset_period is set to 6,
+  ## each 10m new LLD data will be sent, and after 10ms*6=1h all LLDs will be sent again.
+  lld_reset_period = 6
```

Comment:
I have not found any other output with this kind of behavior (having different intervals for internal stuff).

The closest is the `http` input with the `cookie_auth_renewal` option. In this case, it creates a [new goroutine](https://github.com/influxdata/telegraf/blob/3fe151b28d0acaa777cb62868f8ab2623c6759d5/plugins/common/cookie/cookie.go#L44) to periodically renew the cookie.

I can create one goroutine for LLD data and another for the autoregistration and use those _periods_ to trigger the execution.

Another option is changing that values to `config.Duration` and just checking, in each write, if enough time has passed to trigger the executions. But this could be misleading for the users, because that time will be only an approximation, or completely wrong if they try to set these options to a lower value than `flush_interval`.

Reply:
After reading the code I realized that those are "multiples of the gather interval". For me "period" triggers "time", so it can only be me but I think the option name is misleading then. I think having multiples-of-interval makes sense but I would like to see a more intuitive naming like `lld_send_every_nth_interval`or `lld_send_skip_intervals` or something (probably better) along the lines. I think this will reduce confusion and makes it easier for the user to use the plugin.

Reply:
Fixed in 37d78aa67.

Moved to durations, like autoregister. New config options:
- `lld_send_interval`
- `lld_clear_interval`